libopenage/gamestate/player.cpp

```diff
@@ -141,19 +161,56 @@ void Player::active_unit_removed(Unit *unit) {
 		return;
 	}
 
+	this->units_have[unit->unit_type->id()] -= 1;
 	// TODO handle here building dependencies
-	// TODO handle here on death unit triggers
 
 	// population
 	if (unit->has_attribute(attr_type::population)) {
 		auto popul = unit->get_attribute<attr_type::population>();
 		if (popul.demand > 0) {
-			population.free_population(popul.demand);
+			this->population.free_population(popul.demand);
 		}
 		if (popul.capacity > 0) {
-			population.remove_capacity(popul.capacity);
+			this->population.remove_capacity(popul.capacity);
 		}
 	}
+
+	// score
+	// TODO improve selectors
+	if (unit->unit_type->id() == 82 || unit->unit_type->id() == 276) { // Castle, Wonder
```

Comment:
All this information should be inside the UnitType, which I'm not touching because of the upcoming nyan revolution...